src/utils/module.c

```diff
@@ -250,3 +251,83 @@ gboolean bd_utils_unload_kernel_module (const gchar *module_name, GError **error
     freelocale (c_locale);
     return TRUE;
 }
+
+
+static BDUtilsLinuxVersion detected_linux_ver;
+
+G_LOCK_DEFINE_STATIC (detected_linux_ver);
+
+/**
+ * bd_utils_get_linux_version:
+ * @error: (out): place to store error (if any)
+ *
+ * Retrieves version of currently running Linux kernel. Acts also as an initializer for statically cached data.
+ *
+ * Returns: (transfer none): Detected Linux kernel version or %NULL in case of an error. The returned value belongs to the library, do not free.
+ */
+BDUtilsLinuxVersion * bd_utils_get_linux_version (GError **error) {
+    struct utsname buf;
+
+    G_LOCK (detected_linux_ver);
+
+    /* return cached value if available */
+    if (detected_linux_ver.major > 0) {
+        G_UNLOCK (detected_linux_ver);
+        return &detected_linux_ver;
+    }
+
+    if (uname (&buf)) {
+        g_set_error (error, BD_UTILS_MODULE_ERROR, BD_UTILS_MODULE_ERROR_FAIL,
+                     "Failed to get linux kernel version: %m");
+        G_UNLOCK (detected_linux_ver);
+        return NULL;
+      }
+
+    if (g_ascii_strncasecmp (buf.sysname, "Linux", sizeof buf.sysname) != 0) {
+        g_set_error (error, BD_UTILS_MODULE_ERROR, BD_UTILS_MODULE_ERROR_INVALID_PLATFORM,
+                     "Failed to get kernel version: spurious sysname '%s' detected", buf.sysname);
+        G_UNLOCK (detected_linux_ver);
+        return NULL;
+      }
+
+    if (sscanf (buf.release, "%d.%d.%d",
```

Comment:
Please either initialize the struct with zeroes or compare the `sscanf()` return value to `3`.

Reply:
I've added explicit memset and an `initialized` flag for your peaceful sleep in the night :-) (the static initializer over a struct still suffers from that gcc warning bug on RHEL7). I'm still convinced the struct is properly initialized by the compiler via the `static` keyword. And assuming simple data types as `gboolean` are atomic by nature.

As far as checking `sscanf()` result is concerned, it was on purpose to check if at least the first number got matched, in case we ever get a string like `6` instead of `6.0.0`.